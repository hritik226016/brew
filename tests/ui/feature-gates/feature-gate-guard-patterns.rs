// FIXME(guard_patterns): this lint should not be triggered
// once exhaustiveness is implemented correctly
#![allow(irrefutable_let_patterns)]

fn match_guards_still_work() {
    match 0 {
        0 if guard(0) => {},
        _ => {},
    }
}

fn other_guards_dont() {
    match 0 {
        (0 if guard(0)) | 1 => {},
        //~^ ERROR: guard patterns are experimental
        _ => {},
    }

    let ((x if guard(x)) | x) = 0;
    //~^ ERROR: guard patterns are experimental

    if let (x if guard(x)) = 0 {}
    //~^ ERROR: guard patterns are experimental
    while let (x if guard(x)) = 0 {}
    //~^ ERROR: guard patterns are experimental
}

fn even_as_function_parameters(((x if guard(x), _) | (_, x)): (i32, i32)) {}
//~^ ERROR: guard patterns are experimental

fn guard<T>(x: T) -> bool {
    unimplemented!()
}

fn main() {
    match_guards_still_work();
    other_guards_dont();
    even_as_function_parameters((0, 0));
}
