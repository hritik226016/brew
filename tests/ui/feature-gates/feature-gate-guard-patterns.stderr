error[E0658]: guard patterns are experimental
  --> $DIR/feature-gate-guard-patterns.rs:14:10
   |
LL |         (0 if guard(0)) | 1 => {},
   |          ^^^^^^^^^^^^^
   |
   = note: see issue #129967 <https://github.com/rust-lang/rust/issues/129967> for more information
   = help: add `#![feature(guard_patterns)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
   = help: consider using match arm guards

error[E0658]: guard patterns are experimental
  --> $DIR/feature-gate-guard-patterns.rs:19:11
   |
LL |     let ((x if guard(x)) | x) = 0;
   |           ^^^^^^^^^^^^^
   |
   = note: see issue #129967 <https://github.com/rust-lang/rust/issues/129967> for more information
   = help: add `#![feature(guard_patterns)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
   = help: consider using match arm guards

error[E0658]: guard patterns are experimental
  --> $DIR/feature-gate-guard-patterns.rs:22:13
   |
LL |     if let (x if guard(x)) = 0 {}
   |             ^^^^^^^^^^^^^
   |
   = note: see issue #129967 <https://github.com/rust-lang/rust/issues/129967> for more information
   = help: add `#![feature(guard_patterns)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
   = help: consider using match arm guards

error[E0658]: guard patterns are experimental
  --> $DIR/feature-gate-guard-patterns.rs:24:16
   |
LL |     while let (x if guard(x)) = 0 {}
   |                ^^^^^^^^^^^^^
   |
   = note: see issue #129967 <https://github.com/rust-lang/rust/issues/129967> for more information
   = help: add `#![feature(guard_patterns)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
   = help: consider using match arm guards

error[E0658]: guard patterns are experimental
  --> $DIR/feature-gate-guard-patterns.rs:28:34
   |
LL | fn even_as_function_parameters(((x if guard(x), _) | (_, x)): (i32, i32)) {}
   |                                  ^^^^^^^^^^^^^
   |
   = note: see issue #129967 <https://github.com/rust-lang/rust/issues/129967> for more information
   = help: add `#![feature(guard_patterns)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
   = help: consider using match arm guards

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0658`.
